sqla_vertica_python/vertica_python.py

```diff
@@ -65,6 +65,18 @@ def initialize(self, connection):
         super(PGDialect, self).initialize(connection)
         self.implicit_returning = False
 
+    def is_disconnect(self, e, connection, cursor):
+        if isinstance(e, self.dbapi.Error) and connection is not None:
+            try:
+                connection = connection.connection
```

Comment:
I'm a bit confused by this line. `connection` is the underlying `__connection` object as far as I know. There should be no `connection.connection`.

I'm also generally unsure that this implementation is correct. From my understanding, we should just be checking the error object for text like `is closed`, e.g. https://github.com/vertica/vertica-python/blob/master/vertica_python/vertica/connection.py#L247

From [the SQLAlchemy docs](https://docs.sqlalchemy.org/en/latest/core/pooling.html#more-on-invalidation):

> all SQLAlchemy dialects include a system called is_disconnect() which will examine the contents of an exception object, including the string message and any potential error codes included with it, in order to determine if this exception indicates that the connection is no longer usable

Reply:
Sounds good, thanks for the explanation. Made some slight syntactical adjustments. I'll release this tomorrow under the next minor version